chapters/06-smart-contracts-in-ralph.md

```diff
@@ -1405,21 +1403,25 @@ import { OldCode, NewCode } from '../artifacts/ts'
 
 async function test() {
   const signer = await getSigner()
-  const { contractInstance: oldCode } = await OldCode.deploy(signer, {
-     initialFields: { owner: signer.address, n: 100n }
+  const { contractInstance: oldContract } = await OldCode.deploy(signer, {
+    initialFields: { owner: signer.address, n: 100n }
   })
 
-  await oldCode.transact.migrateWithFields({
+  await oldContract.transact.migrateWithFields({
     signer,
     args: { newCode: NewCode.contract.bytecode, newN: 200n }
   })
 
-  const newContract = await NewCode.at(oldCode.address)
-  const {returns: newN} = await newContract.view.get()
+  const newContract = NewCode.at(oldContract.address)
+  let newN = (await newContract.view.get()).returns
   console.assert(newN === 200n)
-  await newContract.transact.set({signer, args: {m: 300n}})
-  const {returns: newN2} = await newContract.view.get()
-  console.assert(newN2 === 300n)
+
+  await newContract.transact.set({ signer, args: { m: 300n } })
+  newN = (await newContract.view.get()).returns
+  console.assert(newN === 300n)
+
+  newN = (await oldContract.view.get()).returns
+  console.assert(newN === 300n)
 
   await newContract.transact.destroy({ signer })
```

Comment:
@h0ngcha0 I am curious, is this line added to demonstrate the destruction of a contract as a concept, or is it part of the flow of migrating the contract? If it's the former, then maybe it's worth its own section rather than mixing it with the migration flow? Just my 2 cents, cause initially it appeared to me that I **need** to delete the contract after I migrate it, as part of the flow, to get back my deposit. But if I delete it, I cannot use the migrated contract anymore. That's why I found it a bit confusing.

> New contract can also destroy itself and return the contract's deposit back to the owner using the destroySelf!() built-in function.

Reply:
If it creates confusion, maybe we can remove it. We could have another example to explain it.

Reply:
I would go for a separate example, but I leave it up to you!